src/mongocxx/include/mongocxx/v_noabi/mongocxx/cursor.hpp

```diff
@@ -40,38 +48,73 @@ namespace v_noabi {
 /// @note By default, cursors timeout after 10 minutes of inactivity.
 ///
 class cursor {
+   private:
+    v1::cursor _cursor;
+    bsoncxx::v_noabi::document::view _doc;
+
    public:
-    enum class type { k_non_tailable, k_tailable, k_tailable_await };
+    using type = v1::cursor::type;
 
     class iterator;
 
     ///
     /// Move constructs a cursor.
     ///
-    MONGOCXX_ABI_EXPORT_CDECL() cursor(cursor&&) noexcept;
+    MONGOCXX_ABI_EXPORT_CDECL() cursor(cursor&& other) noexcept;
 
     ///
     /// Move assigns a cursor.
     ///
-    MONGOCXX_ABI_EXPORT_CDECL(cursor&) operator=(cursor&&) noexcept;
+    MONGOCXX_ABI_EXPORT_CDECL(cursor&) operator=(cursor&& other) noexcept;
 
     ///
     /// Destroys a cursor.
     ///
     MONGOCXX_ABI_EXPORT_CDECL() ~cursor();
 
-    cursor(cursor const&) = delete;
-    cursor& operator=(cursor const&) = delete;
+    ///
+    /// This class is not copyable.
+    ///
+    cursor(cursor const& other) = delete;
+    ///
+    /// This class is not copyable.
+    ///
+    cursor& operator=(cursor const& other) = delete;
+
+    ///
+    /// Construct with the @ref mongocxx::v1 equivalent.
+    ///
+    /* explicit(false) */ cursor(v1::cursor cursor) : _cursor{std::move(cursor)} {}
+
+    ///
+    /// Convert to the @ref bsoncxx::v1 equivalent.
+    ///
+    /// @par Postconditions:
+    /// - `other` is in an assign-or-destroy-only state.
+    ///
+    /// @warning Invalidates all associated iterators and views.
+    ///
+    explicit operator v1::cursor() && {
+        return std::move(_cursor);
+    }
+
+    ///
+    /// This class is not copyable.
+    ///
+    explicit operator v1::cursor() const& = delete;
 
     ///
-    /// A cursor::iterator that points to the beginning of any available
-    /// results.  If begin() is called more than once, the cursor::iterator
-    /// returned points to the next remaining result, not the result of
-    /// the original call to begin().
+    /// A cursor::iterator points to the beginning of any available results.
     ///
-    /// For a tailable cursor, when cursor.begin() == cursor.end(), no
-    /// documents are available.  Each call to cursor.begin() checks again
-    /// for newly-available documents.
+    /// The first call to begin() advances to the next available document. Consecutive calls to begin() only advance  to
```

Comment:
Remove extra whitespace:

```suggestion
    /// The first call to begin() advances to the next available document. Consecutive calls to begin() only advance to
```